website/content/docs/enterprise/pkcs11-provider/index.mdx

```diff
@@ -0,0 +1,301 @@
+
+| Name               | Mechanism Number |
+| ------------------ | ---------------- |
+| AES key generation | `0x1080`         |
+| AES-CBC            | `0x1082`         |
+| AES-CBC Pad        | `0x1085`         |
+| AES-CTR            | `0x1086`         |
+| AES-GCM            | `0x1087`         |
+
+Here is the list of supported and unsupported PKCS#11 functions:
+
+- Encryption and decryption
+  - [x]  `C_EncryptInit`
+  - [x]  `C_Encrypt`
+  - [ ]  `C_EncryptUpdate`
+  - [ ]  `C_EncryptFinal`
+  - [x]  `C_DecryptInit`
+  - [x]  `C_Decrypt`
+  - [ ]  `C_DecryptUpdate`
+  - [ ]  `C_DecryptFinal`
+- Key management
+  - [x]  `C_GenerateKey`
+  - [ ]  `C_GenerateKeyPair`
+  - [ ]  `C_WrapKey`
+  - [ ]  `C_UnwrapKey`
+  - [ ]  `C_DeriveKey`
+- Objects
+  - [x]  `C_CreateObject`
+  - [x]  `C_DestroyObject`
+  - [x]  `C_GetAttributeValue`
+  - [x]  `C_FindObjectsInit`
+  - [x]  `C_FindObjects`
+  - [x]  `C_FindObjectsFinal`
+  - [ ]  `C_SetAttributeValue`
+  - [ ]  `C_CopyObject`
+  - [ ]  `C_GetObjectSize`
+- Management
+  - [x]  `C_Initialize`
+  - [x]  `C_Finalize`
+  - [x]  `C_Login` (PIN is used as a passphrase for the TLS encryption key, if provided)
+  - [x]  `C_Logout`
+  - [x]  `C_GetInfo`
+  - [x]  `C_GetSlotList`
+  - [x]  `C_GetSlotInfo`
+  - [x]  `C_GetTokenInfo`
+  - [x]  `C_GetMechanismList`
+  - [x]  `C_GetMechanismInfo`
+  - [x]  `C_OpenSession`
+  - [x]  `C_CloseSession`
+  - [x]  `C_CloseAllSessions`
+  - [x]  `C_GetSessionInfo`
+  - [ ]  `C_InitToken`
+  - [ ]  `C_InitPIN`
+  - [ ]  `C_SetPIN`
+  - [ ]  `C_GetOperationState`
+  - [ ]  `C_SetOperationState`
+  - [ ]  `C_GetFunctionStatus`
+  - [ ]  `C_CancelFunction`
+  - [ ]  `C_WaitForSlotEvent`
+- Signing
+  - [ ]  `C_SignInit`
+  - [ ]  `C_Sign`
+  - [ ]  `C_SignUpdate`
+  - [ ]  `C_SignFinal`
+  - [ ]  `C_SignRecoverInit`
+  - [ ]  `C_SignRecover`
+  - [ ]  `C_VerifyInit`
+  - [ ]  `C_Verify`
+  - [ ]  `C_VerifyUpdate`
+  - [ ]  `C_VerifyFinal`
+  - [ ]  `C_VerifyRecoverInit`
+  - [ ]  `C_VerifyRecover`
+- Digests
+  - [ ]  `C_DigestInit`
+  - [ ]  `C_Digest`
+  - [ ]  `C_DigestUpdate`
+  - [ ]  `C_DigestKey`
+  - [ ]  `C_DigestFinal`
+  - [ ]  `C_DigestEncryptUpdate`
+  - [ ]  `C_DecryptDigestUpdate`
+  - [ ]  `C_SignEncryptUpdate`
+  - [ ]  `C_DecryptVerifyUpdate`
+- Random Number Generation
+  - [ ]  `C_SeedRandom`
+  - [ ]  `C_GenerateRandom`
+
+## Limitations and Notes
+
+Due to the nature of Vault, the KMIP Secrets Engine, and PKCS#11, there are some other limitations to be aware of:
+
+- The key and object IDs returned by `C_FindObjects`, etc., are randomized for each session, and cannot be shared between sessions and have no meaning after a session is closed.
+  This is because KMIP objects, which are used to store the PKCS#11 objects, have long random strings as IDs, but the PKCS#11 object ID is limited to a 32-bit integer, in addition to the fact that the PKCS#11 provider does not have any local storage.
```

Comment:
```suggestion
  This is because KMIP objects, which are used to store the PKCS#11 objects, have long random strings as IDs, but the PKCS#11 object ID is limited to a 32-bit integer. Also, the PKCS#11 provider does not have any local storage.
```